Hermes/appscale/hermes/stats/subscribers/cache.py

```diff
@@ -0,0 +1,153 @@
+import threading
+import time
+from datetime import datetime
+
+from appscale.hermes.stats.pubsub_base import StatsSubscriber
+
+BIG_BANG_TIMESTAMP = 0
+
+
+class StatsCache(StatsSubscriber):
+  """
+  It takes care about storing snapshots in limited cache and provides
+  reading method with acknowledgment mechanism.
+  Each node has local stats cache for each kind of stats it collects
+  (node stats, processes stats and haproxy stats for LB nodes).
+  It is used as a temporary storage for stats which wasn't read by master yet.
+  """
+
+  def __init__(self, snapshots_cache_size, ttl=None):
+    self._snapshots_cache = []
+    self._cache_lock = threading.Lock()
```

Comment:
No, everything uses the same IOLoop (unless you explicitly create multiple IOLoops in different threads, but that would be unusual). It's one thing that makes async programming nice because you don't have to worry about some types of race conditions. It also makes it important not to block the loop.